url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/data/Generic_UKPostcode_Handler.java

```diff
@@ -0,0 +1,428 @@
+        } else {
+            result[0] = 2;
+            result[1] = "Incorrect form of firstPartPostcode (" + firstPartPostcode + ")."
+                    + " Correct form of secondPartPostcode (" + secondPartPostcode + "): " + secondPartPostcodeType + ".";
+        }
+        return result;
+    }
+
+    /**
+     * Check SecondPart Postcode is in the format "NAA", "NA" or "N" where a
+     * stands for an alphabetical character and n stands for a numerical
+     * character.
+     *
+     * @param secondPartPostcode
+     * @return
+     */
+    public String getSecondPartPostcodeType(String secondPartPostcode) {
+        if (secondPartPostcode.length() < 4) {
+            switch (secondPartPostcode.length()) {
+                case 3: {
+                    String _0 = secondPartPostcode.substring(0, 1);
+                    String _1 = secondPartPostcode.substring(1, 2);
+                    String _2 = secondPartPostcode.substring(2, 3);
+                    if (get_0to9().contains(_0)
+                            && get_AtoZ_not_CIKMOV().contains(_1)
+                            && get_AtoZ_not_CIKMOV().contains(_2)) {
+                        return sNAA;
+                    }
+                    break;
+                }
+                case 2: {
+                    String _0 = secondPartPostcode.substring(0, 1);
+                    String _1 = secondPartPostcode.substring(1, 2);
+                    if (get_0to9().contains(_0)
+                            && get_AtoZ_not_CIKMOV().contains(_1)) {
+                        return sNA;
+                    }
+                    break;
+                }
+                case 1: {
+                    String _0 = secondPartPostcode.substring(0, 1);
+                    if (get_0to9().contains(_0)) {
+                        return sN;
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+        return s;
+    }
+
+    public HashSet<String> firstPartPostcodeTypes;
+
+    public HashSet<String> getFirstPartPostcodeTypes() {
+        if (firstPartPostcodeTypes == null) {
+            firstPartPostcodeTypes = new HashSet<String>();
+            firstPartPostcodeTypes.add(sAN);
+            firstPartPostcodeTypes.add(sANN);
+            firstPartPostcodeTypes.add(sAAN);
+            firstPartPostcodeTypes.add(sAANN);
+            firstPartPostcodeTypes.add(sANA);
+            firstPartPostcodeTypes.add(sAANA);
+        }
+        return firstPartPostcodeTypes;
+    }
+
+    /**
+     * If firstPartPostcode is not a valid first part for a postcode return an
+     * empty string otherwise return "AANN", "AANA", "ANN", "ANA", "AAN", "AN"
+     * respectively where A stands for an alphabetical character and N stands
+     * for a numerical character.
+     *
+     * @param firstPartPostcode
+     * @return
+     */
+    public String getFirstPartPostcodeType(String firstPartPostcode) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 304-312 into a separate method.